protos/feast/core/FeatureSource.proto

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright 2020 The Feast Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+
+syntax = "proto3";
+package feast.core;
+
+option go_package = "github.com/feast-dev/feast/sdk/go/protos/feast/core";
+option java_outer_classname = "FeatureSourceProto";
+option java_package = "feast.proto.core";
+
+// Defines a Feature Source that can be used source Feature data
+message FeatureSourceSpec {
+  // Type of Feature Source.
+  enum SourceType {
+    INVALID = 0;
+    BATCH_FILE = 1;
+    BATCH_BIGQUERY = 2;
+    STREAM_KAFKA = 3;
+    STREAM_KINESIS = 4;
+  }
+  SourceType type = 1;
+
+  // Defines mapping between fields in the sourced data 
+  // and fields in parent FeatureTable.
+  map<string, string> field_mapping = 2;
+
+  // Must specify timestamp column name
+  string ts_column = 3;
+
+  // (Optional) Specify partition column
+  // useful for file sources
+  string date_partition_column = 4;
+
+  // Defines options for Feature source that sources features from a file
+  message FileOptions {
+    // File Format of the file containing the features
+    enum FileFormat {
+       INVALID = 0;
+       PARQUET = 1;
+       AVRO = 2;
+    }
+    FileFormat file_format = 1;
+
+    // Target URL of file to retrieve and source features from.
+    // s3://path/to/file for AWS S3 storage
+    // gs://path/to/file for GCP GCS storage
+    // file:///path/to/file for local storage
+    string file_url  = 2;
+  }
+
+  // Defines options for Feature source that sources features from a BigQuery Query
+  message BigQueryOptions {
+    // GCP Project ID
+    string project_id = 1;
+
+    // SQL Query used by BQ Source to query feature data.
+    string sql_query = 2;
```

Comment:
Spark BQ Connector doesn't support arbitrary sql query. So we should change this to table reference instead. You can refer to the updated Feast 0.8 RFC.

Reply:
Updated proto to use reference.